### PR TITLE
Add subscriptions for buttons

### PR DESCRIPTION
**Pull request by stevenewald**

### CODE REVIEW

File: external/microbit_v2/Board.mk [hunk 1]
```diff
@@ -60,7 +60,6 @@ BOARD_SOURCES += \
 	nrf_strerror.c\
 	nrf_twi_mngr.c\
 	nrfx_clock.c\
-	nrfx_gpiote.c\
```
Comment (stevenewald) [inline external/microbit_v2/Board.mk:63]: nrfx gpiote implements its own handler, which causes linker errors. We don't want nrfx anyway (thats why we have edge::aidan)

Comment (echavemann) [reply external/microbit_v2/Board.mk:63]: nice

File: include/config.hpp
Comment (echavemann) [inline include/config.hpp:1]: good call, is this formatted?

File: include/drivers/buttons.hpp [hunk 2]
```diff
@@ -1,24 +1,52 @@
...
 
-#include "drivers/button_type.hpp"
+#include "config.hpp"
+#include "driver_enums.hpp"
 #include "drivers/gpio_pin.hpp"
+#include "drivers/gpiote_pin.hpp"
 #include "microbit_v2.h"
 
 namespace edge::drivers {
+// Without runtime std::function, we can't directly point the gpio callback to
```
Comment (echavemann) [inline include/drivers/buttons.hpp:10]: this is quite cursed but maybe ok.

**Review by echavemann — APPROVED** [review]
i likeit

**Pull request merged**